internal/tools/semconv-gen/generator.go

```diff
@@ -0,0 +1,300 @@
+	}
+
+	err = exec.Command("cp", "-a", cfg.input, path.Join(tmpDir, "input")).Run()
+	if err != nil {
+		return fmt.Errorf("unable to copy input to temp directory: %w", err)
+	}
+
+	err = exec.Command("cp", cfg.template, tmpDir).Run()
+	if err != nil {
+		return fmt.Errorf("unable to copy template to temp directory: %w", err)
+	}
+
+	cmd := exec.Command("docker", "run", "--rm",
+		"-v", fmt.Sprintf("%s:/data", tmpDir),
+		cfg.image,
+		"--yaml-root", path.Join("/data/input", path.Base(cfg.input)),
+		"code",
+		"--template", path.Join("/data", path.Base(cfg.template)),
+		"--output", path.Join("/data/output", path.Base(cfg.target)),
+	)
+	err = cmd.Run()
+	if err != nil {
+		return fmt.Errorf("unable to render template: %w", err)
+	}
+
+	err = exec.Command("cp", path.Join(tmpDir, "output", path.Base(cfg.target)), cfg.output).Run()
+	if err != nil {
+		return fmt.Errorf("unable to copy result to target: %w", err)
+	}
+
+	return nil
+}
+
+func findRepoRoot() (string, error) {
+	start, err := os.Getwd()
+	if err != nil {
+		return "", err
+	}
+
+	dir := start
+	for {
+		_, err := os.Stat(filepath.Join(dir, ".git"))
+		if errors.Is(err, os.ErrNotExist) {
+			dir = filepath.Dir(dir)
+			// From https://golang.org/pkg/path/filepath/#Dir:
+			// The returned path does not end in a separator unless it is the root directory.
+			if strings.HasSuffix(dir, string(filepath.Separator)) {
+				return "", fmt.Errorf("unable to find git repository enclosing working dir %s", start)
+			}
+			continue
+		}
+
+		if err != nil {
+			return "", err
+		}
+
+		return dir, nil
+	}
+}
+
+var commonInitialisms = []string{
+	"ACL",
+	"AIX",
+	"AKS",
+	"AMD64",
+	"API",
+	"ARM32",
+	"ARM64",
+	"ARN",
+	"ASCII",
+	"AWS",
+	"CPU",
+	"CSS",
+	"DB",
+	"DNS",
+	"EC2",
+	"ECS",
+	"EDB",
+	"EKS",
+	"EOF",
+	"GCP",
+	"GUID",
+	"HPUX",
+	"HSQLDB",
+	"HTML",
+	"HTTP",
+	"HTTPS",
+	"IA64",
+	"ID",
+	"IP",
+	"JDBC",
+	"JSON",
+	"K8S",
+	"LHS",
+	"MSSQL",
+	"OS",
+	"PHP",
+	"PID",
+	"PPC32",
+	"PPC64",
+	"QPS",
+	"QUIC",
+	"RAM",
+	"RHS",
+	"RPC",
+	"SDK",
+	"SLA",
+	"SMTP",
+	"SPDY",
+	"SQL",
+	"SSH",
+	"TCP",
+	"TLS",
+	"TTL",
+	"UDP",
+	"UID",
+	"UI",
+	"UUID",
+	"URI",
+	"URL",
+	"UTF8",
+	"VM",
+	"XML",
+	"XMPP",
+	"XSRF",
+	"XSS",
+	"ZOS",
+}
+
+var replacements = map[string]string{
+	"Mysql":         "MySQL",
+	"Postgresql":    "PostgreSQL",
+	"Mariadb":       "MariaDB",
+	"Maxdb":         "MaxDB",
+	"Firstsql":      "FirstSQL",
+	"Instantdb":     "InstantDB",
+	"RedisDatabase": "RedisDB",
+	"Hbase":         "HBase",
+	"Mongodb":       "MongoDB",
+	"Couchdb":       "CouchDB",
+	"Cosmosdb":      "CosmosDB",
+	"Dynamodb":      "DynamoDB",
+	"Freebsd":       "FreeBSD",
+	"Netbsd":        "NetBSD",
+	"Openbsd":       "OpenBSD",
+	"Dragonflybsd":  "DragonflyBSD",
+	"Inproc":        "InProc",
+	"IPTCP":         "TCP",
+	"IPUDP":         "UDP",
+	"Faas":          "FaaS",
+	"Lineno":        "LineNumber",
+}
+
+func fixInitialisms(fn string) error {
```

Comment:
there is quite a logic here 😉 
what do you think of adding some tests here using some input and expected "golden files"  kept in `testdata` directory?
It would help fix potential problems with the replacement mechanism in a more automated and faster way.

Reply:
The fixup logic is pretty simple, I think.  I've added some comments and separated the `go fmt` invocation.  I view the spec as the "golden files" and the current output as the target for testing any refactoring.  It should always be possible to remove the generated output, regenerate them, and check that git doesn't think anything has changed if testing a refactoring.  If testing a change intended to induce a change in the output, the same thing applies.  Git's diff mechanism can let us know if we've done what we expected to do, though we shouldn't be changing generated outputs once they're shipped so I don't think it will be an issue in practice.